### PR TITLE
Check user's plan and repo privacy for repo rules

### PR DESCRIPTION
**Pull request by vaindil**
Part of #16707

## Description
Repo rules API endpoints return a 403 for private repos if the owner is on the free plan, as free users can only use them in public repos. This adds a check for that to avoid these errors from being thrown.

### Screenshots

N/A

## Release notes

Notes: no-notes

### CODE REVIEW

File: app/src/lib/helpers/repo-rules.ts [hunk 1]
```diff
@@ -12,6 +12,55 @@ import {
...
+import { Account } from '../../models/account'
+import { Repository } from '../../models/repository'
+import { GitHubRepository } from '../../models/github-repository'
+
+/**
+ * Returns whether repo rules could potentially exist for the provided account and repository.
+ * This only performs client-side checks, such as whether the user is on a free plan
+ * and the repo is public.
+ */
+export function useRepoRulesLogic(
```
Comment (tidy-dev) [inline app/src/lib/helpers/repo-rules.ts:28]: I think we can simplify this method a bit by making changing the repository parameter to `repository: Repository` and using the `isRepositoryWithGitHubRepository` help method to identify if the repo has a GitHubRepository object on it.

Here shows what I am talking about:
https://github.com/desktop/desktop/compare/40017cac396ce76558ebb70a8724be7b6e765203...Use-isRepositoryWithGitHubRepository-Helper

Comment (vaindil) [reply app/src/lib/helpers/repo-rules.ts:28]: Good call, I didn't know about that helper, thanks! Just pushed that.

Comment (Aipasa) [reply app/src/lib/helpers/repo-rules.ts:28]: 2e5f7947e18e178a0494f28f1ccb7bc798a5aafa

**Review by tidy-dev — APPROVED** [review]
Looks good! Thanks for the change.

**Pull request merged**